lib/OcrProcessors/PdfOcrProcessor.php

```diff
@@ -48,20 +50,31 @@ class PdfOcrProcessor implements IOcrProcessor {
 	/** @var ICommand */
 	private $command;
 
+	/** @var IImageToPdfConverter */
+	private $converter;
+
 	/** @var LoggerInterface */
 	private $logger;
 
-	public function __construct(ICommand $command, LoggerInterface $logger) {
+	public function __construct(ICommand $command, IImageToPdfConverter $converter, LoggerInterface $logger) {
 		$this->command = $command;
+		$this->converter = $converter;
 		$this->logger = $logger;
 	}
 
-	public function ocrFile(string $fileContent, WorkflowSettings $settings, GlobalSettings $globalSettings): string {
+	public function ocrFile(File $file, WorkflowSettings $settings, GlobalSettings $globalSettings): OcrProcessorResult {
+		if ($file->getMimeType() !== 'application/pdf') {
+			// Convert file to pdf. Here we assume that we're dealing with an image input
+			$pdfContent = $this->converter->convertToPdf($file->getContent());
```

Comment:
Why do we need a conversion from an image file to a PDF file here?

If we convert an image to a PDF and pass the PDF to OCRmyPDF, the original input image will be converted twice. OCRmyPDF converts the input PDF again (using ghostscript) before feeding tesseract with the double converted image. Due to the 2-step conversion of input images, the image quality suffers with lossy input image formats (e.g. JPEG) which could result in a degraded OCR quality. So, why do we not remove any image conversion and pass the input images directly to tesseract instead?